### ISSUE TITLE
report_lint matcher: Failed match of specific count of lints prints a useless message

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#524]
In the linter specs, we can specify that we expect an exact number of lints on a line (this is used a number of times; `git grep 'line: [0-9]*, count: [0-9]*' spec/scss_lint/linter`), but when it fails the message is not useful.

For example, if we change border_zero_spec.rb, line 41, to: `it { should report_lint line: 2, count: 2 }`, then we get the following failure from RSpec:

```
1) SCSSLint::Linter::BorderZero when a property has a border of none should report a lint on line 2
   Failure/Error: it { should report_lint line: 2, count: 2 }
     expected that a lint would be reported on line 2, but was on line 2
   # ./spec/scss_lint/linter/border_zero_spec.rb:41:in `block (4 levels) in <top (required)>'
```

That is, "expected that a lint would be reported on line 2, but was on line 2".

**Comment by sds** [comment]
Implemented in #525.
